src/pygenstability/data_clustering.py

```diff
@@ -0,0 +1,286 @@
+"""Construct geometric graphs from data for multiscale clustering."""
+
+import numpy as np
+
+from scipy.spatial.distance import pdist, squareform
+from scipy.sparse.csgraph import minimum_spanning_tree
+from scipy.sparse import csr_matrix
+
+from pygenstability.pygenstability import run as pgs_run
+from pygenstability.plotting import plot_scan as pgs_plot_scan
+from pygenstability.optimal_scales import identify_optimal_scales
+
+
+def compute_kNN(D, k=5):
```

Comment:
you could use this here: from sklearn.neighbors import kneighbors_graph

Reply:
done